bindings/go/oci/repository.go

```diff
@@ -99,8 +99,9 @@ func (repo *Repository) AddComponentVersion(ctx context.Context, descriptor *des
 		return fmt.Errorf("failed to add descriptor to store: %w", err)
 	}
 
-	// Tag the manifest with the reference
-	if err := store.Tag(ctx, *manifest, version); err != nil {
+	// Tag the manifest with the version, but first remove prohibited characters.
+	tag := LooseSemverToOCITag(version)
+	if err := store.Tag(ctx, *manifest, tag); err != nil {
```

Comment:
why dont you just use the reference here?
```suggestion
	if err := store.Tag(ctx, *manifest, reference); err != nil {
```

Reply:
I don't use `reference` here, because you were not using it in the first place. And it look like you did it for a reason, because if replaced with `reference`, the CTF tests start to fail.

```
        	Error:      	Received unexpected error:
        	            	failed to get manifest: failed to resolve reference "ctf.ocm.software/component-descriptors/test-component:v1.0.0": not found
        	Test:       	Test_Integration_CTF/direct/basic_upload_and_download_of_a_component_version
--- FAIL: Test_Integration_CTF/direct/basic_upload_and_download_of_a_component_version (0.01s)
```

So, do we want to keep the tag/version here?

Reply:
This seems to be buggy behavior. you should probably check out why this is failing because the reference should work.

Reply:
Seems to be fixed. Ptal.